### DISCUSSION TITLE
Experiments with speech to text and novelWriter

### DISCUSSION THREAD

**Discussion by dgoadby** [vkbo/novelWriter#1413] [Ideas]
I love novelWriter but I am a pi$$ poor typist and it is slowing down my "writers flow".  

I already have an earlier version of Dragon but it is Windows only and I didn't really get on with it. 

There are a couple of direct speech to text applications that are looking quite usable (and free) which are Whisper and Vosk. I use Linux mainly and I will create containerised versions of the above to see how well they can be integrated into novelWriter. 

They will not be suitable for users with low power PC's and laptops as there is quite a large memory and processing speed requirement but I think it is worth a try on my hardware. 

If there is interest I am happy to share the results here. 

Is anyone else playing around with speech to text?

**Comment by vkbo** [comment]
I haven't given this any thought myself, so it would be interesting to see the result.

Do these tools need to be integrated in any way? Or do they just work like a virtual keyboard?

**Reply by dgoadby** [reply]
My plan is to pipe the text output in a way that looks like a virtual keyboard. 

The idea is the real keyboard will still function for error correction, formatting etc. Once functional, it should be fairly easy to incorporate novelWriter commands such as "new paragraph" using a trigger word followed by the command.  

My headset arrives tomorrow so we shall see where this goes. 

My carpal tunnels will certainly appreciate it if the project works!
